src/components/decimal/decimal.component.js

```diff
@@ -11,247 +16,261 @@ import LocaleContext from "../../__internal__/i18n-context";
-          shouldCallOnChange = value !== this.emptyValue;
-          return {
-            value: this.emptyValue,
-            visibleValue: this.formatValue(this.emptyValue),
-          };
-        }
-        return {
-          visibleValue: this.formatValue(value),
-        };
-      },
-      () => {
-        if (shouldCallOnChange) {
-          this.callOnChange();
-        }
-        if (this.props.onBlur) {
-          this.props.onBlur(this.createEvent());
-        }
-      }
-    );
-  };
-
-  createEvent = () => {
-    const standardVisible = this.toStandardDecimal(this.state.visibleValue);
-    const formattedValue = this.isNaN(standardVisible)
-      ? this.state.visibleValue
-      : this.formatValue(standardVisible);
-    return {
-      target: {
-        name: this.props.name,
-        id: this.props.id,
-        value: {
-          rawValue: this.state.value,
-          formattedValue,
-        },
-      },
-    };
-  };
-
-  /**
-   * Determine if the component is controlled at the time of call
-   */
-  isControlled() {
-    return this.props.value !== undefined;
-  }
-
-  isNaN = (value) => {
-    return Number.isNaN(Number(value));
-  };
-
-  getSafeValueProp = (isInitialValue) => {
-    const { value, allowEmptyValue } = this.props;
+  const getSafeValueProp = (isInitialValue) => {
     // We're intentionally preventing the use of number values to help prevent any unintentional rounding issues
     invariant(
-      typeof value === "string",
+      typeof isInitialValue === "string",
       "Decimal `value` prop must be a string"
     );
 
     if (isInitialValue && !allowEmptyValue) {
       invariant(
-        value !== "",
+        isInitialValue !== "",
         "Decimal `value` must not be an empty string. Please use `allowEmptyValue` or `0.00`"
       );
     }
-    return value;
+    return isInitialValue;
   };
 
-  removeDelimiters = (value) => {
-    const delimiterMatcher = new RegExp(
-      `[\\${this.getSeparator("group")} ]*`,
-      "g"
-    );
-    return value.replace(delimiterMatcher, "");
-  };
+  const getSeparator = useCallback(
+    (separatorType) => {
+      const numberWithGroupAndDecimalSeparator = 10000.1;
+      return Intl.NumberFormat(locale || l.locale())
+        .formatToParts(numberWithGroupAndDecimalSeparator)
+        .find((part) => part.type === separatorType).value;
+    },
+    [l, locale]
+  );
+
+  const isNaN = useCallback((valueToTest) => {
+    return Number.isNaN(Number(valueToTest));
+  }, []);
 
   /**
    * Format a user defined value
    */
-  formatValue = (value) => {
-    if (this.isNaN(value)) {
-      return value;
-    }
+  const formatValue = useCallback(
+    (valueToFormat) => {
+      if (isNaN(valueToFormat)) {
+        return valueToFormat;
+      }
 
-    /* Guards against any white-space only strings like "   " being 
+      /* Guards against any white-space only strings like "   " being 
        mishandled and returned as `NaN` for visibleValue */
```

Comment:
`visibleValue` no longer exists so this comment needs to change

Reply:
Good spot. I'll get this amended.